src/simplewallet/simplewallet.cpp

```diff
@@ -3982,51 +4146,275 @@ void simple_wallet::stop()
+  {
+    // print all the existing accounts
+    LOCK_IDLE_SCOPE();
+    print_accounts();
+    return true;
+  }
+
+  std::vector<std::string> local_args = args;
+  std::string command = local_args[0];
+  local_args.erase(local_args.begin());
+  if (command == "new")
+  {
+    // create a new account and switch to it
+    std::string label = boost::join(local_args, " ");
+    if (label.empty())
+      label = tr("(Untitled account)");
+    m_wallet->add_subaddress_account(label);
+    m_current_subaddress_account = m_wallet->get_num_subaddress_accounts() - 1;
+    update_prompt();
+    LOCK_IDLE_SCOPE();
+    print_accounts();
+  }
+  else if (command == "switch" && local_args.size() == 1)
+  {
+    // switch to the specified account
+    uint32_t index_major;
+    if (!epee::string_tools::get_xtype_from_string(index_major, local_args[0]))
+    {
+      fail_msg_writer() << tr("failed to parse index: ") << local_args[0];
+      return true;
+    }
+    if (index_major >= m_wallet->get_num_subaddress_accounts())
+    {
+      fail_msg_writer() << tr("specify an index between 0 and ") << (m_wallet->get_num_subaddress_accounts() - 1);
+      return true;
+    }
+    m_current_subaddress_account = index_major;
+    update_prompt();
+    show_balance();
+  }
+  else if (command == "label" && local_args.size() >= 1)
+  {
+    // set label of the specified account
+    uint32_t index_major;
+    if (!epee::string_tools::get_xtype_from_string(index_major, local_args[0]))
+    {
+      fail_msg_writer() << tr("failed to parse index: ") << local_args[0];
+      return true;
+    }
+    local_args.erase(local_args.begin());
+    std::string label = boost::join(local_args, " ");
+    try
+    {
+      m_wallet->set_subaddress_label({index_major, 0}, label);
+      LOCK_IDLE_SCOPE();
+      print_accounts();
+    }
+    catch (const std::exception& e)
+    {
+      fail_msg_writer() << e.what();
+    }
+  }
+  else
+  {
+    fail_msg_writer() << tr("usage: account [new <label text with white spaces allowed> | switch <index> | label <index> <label text with white spaces allowed>]");
+  }
+  return true;
+}
+//----------------------------------------------------------------------------------------------------
+void simple_wallet::update_prompt()
+{
+  std::ostringstream prompt;
+  prompt << "[" << tr("wallet/") << m_current_subaddress_account << " " << m_wallet->get_subaddress_as_str({m_current_subaddress_account, 0}).substr(0, 6) << "]: ";
+  m_cmd_binder.set_prompt(prompt.str());
+}
+//----------------------------------------------------------------------------------------------------
+void simple_wallet::print_accounts()
+{
+  success_msg_writer() << boost::format("%15s %21s %21s %21s") % tr("Account") % tr("Balance") % tr("Unlocked balance") % tr("Label");
+  for (uint32_t account_index = 0; account_index < m_wallet->get_num_subaddress_accounts(); ++account_index)
+  {
+    success_msg_writer() << boost::format(tr("%8u %6s %21s %21s %21s"))
```

Comment:
change a bit for unicode support
`success_msg_writer() << boost::format(tr("%8u %6s %21s %21s %21u")...`